recipes/readline/all/conanfile.py

```diff
@@ -0,0 +1,91 @@
+import os
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class ReadLineConan(ConanFile):
+    name = "readline"
+    description = "A set of functions for use by applications that allow users to edit command lines as they are typed in"
+    topics = ("conan", "readline", "cli", "terminal", "command")
+    license = "GPL-3.0-only"
+    homepage = "https://tiswww.case.edu/php/chet/readline/rltop.html"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = "patches/**"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_library": ["termcap", "curses"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_library": "termcap",
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_library == "termcap":
+            self.requires("termcap/1.3.1")
+        elif self.options.with_library == "curses":
+            self.requires("ncurses/6.2")
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("readline does not support Visual Studio")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("readline-{}".format(self.version), self._source_subfolder)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+
+        configure_args = [
+            "--with-curses={}".format("yes" if self.options.with_library == "curses" else "no"),
+        ]
+        if self.options.shared:
+            configure_args.extend(["--enable-shared", "--disable-static"])
+        else:
+            configure_args.extend(["--enable-static", "--disable-shared"])
+        if tools.cross_building(self.settings):
+            configure_args.append("bash_cv_wcwidth_broken=yes")
```

Comment:
interesting configuration ...

Reply:
I copied it from https://github.com/bincrafters/conan-readline/blob/943768bbcc13e6ef468e4db0efe86933119a2312/conanfile.py#L46-L47

This will override a test that would run on the build system, which would fail to run when cross building.
I'm not totally convinced that this override is really required because the configure script has a test for `$cross_compiling` in that test.

But anyways, I let those 2 lines in.